src/content/blog/astro-480.mdx

```diff
@@ -0,0 +1,193 @@
+    </button>
+  );
+}
+
+export function Comment({ postId }: { postId: string }) {
+  return (
+    <form
+      onSubmit={async (e) => {
+        e.preventDefault();
+        const formData = new FormData(e.target);
+        const result = await actions.blog.comment(formData);
+        // handle result
+      }}
+    >
+      <input type="hidden" name="postId" value={postId} />
+      <label for="author">Author</label>
+      <input id="author" type="text" name="author" />
+      <textarea rows={10} name="body"></textarea>
+      <button type="submit">Post</button>
+    </form>
+  );
+}
+```
+
+For more information on Astro actions, visit [TODO: link to docs].
+
+## Experimental: Rewriting support
+
+Astro 4.8 introduces experimental support for rewriting. Rerouting allows you to show the content of a different page from the current URL. This can be useful for showing different content based on the user's location, device, or other conditions, without needing multiple different URLs. This feature is sometimes also called "rerouting" in other frameworks.
+
+To enable this feature, enable the `experimental.rewriting` option in your Astro config:
+
+```js
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+  experimental: {
+    rewriting: true
+  }
+})
+```
+
+From your pages, you can then access a `rewrite` method on the Astro global or context object:
+
+```astro
+---
+if (!Astro.props.allowed) {
+	return Astro.rewrite("/")
+}
+---
+```
+
+```ts
+export function GET(ctx) {
+	if (!ctx.locals.allowed) {
+		return ctx.rewrite("/")
+	}
+}
+```
+
+This feature can also be used inside middlewares, by passing a parameter to the `next()` method with the same type as the `rewrite` method.
+
+```ts
+export function onRequest(ctx, next) {
+	if (!ctx.cookies.get("allowed")) {
+		return next("/")
+	}
+	return next();
+}
+```
+
+For more information on rewriting, visit [TODO: link to docs].
+
+## Performance improvements
+
+Thanks to the recent hard work from [@bluwy](https://github.com/bluwy) both across the JavaScript ecosystem and in Astro itself, Astro is faster than ever! This release includes a number of performance improvements, improving build and rendering times across the board.
+
+In our own benchmarks, we've observed build times improve by up to 20% in specific cases. No changes are needed to take advantage of these improvements, just upgrade to Astro 4.8 and enjoy the faster builds!
+
+## Ability to define multiple routes with the same entrypoint
+
+Previously, trying to inject multiple routes with the same entrypoint using the [`injectRoute`](https://docs.astro.build/en/reference/integrations-reference/#injectroute-option) function would result in an unintended behavior at build time where the last injected route would override the previous ones. This issue has existed since the introduction of the `injectRoute` function, almost two years ago (in Astro 1.0), and is now finally fixed in Astro 4.8.
```

Comment:
```suggestion
Astro integration authors discovered that trying to inject multiple routes with the same entrypoint using the [`injectRoute`](https://docs.astro.build/en/reference/integrations-reference/#injectroute-option) function would result in an unintended behavior at build time where the last injected route would override the previous ones. This issue has existed since the introduction of the `injectRoute` function, almost two years ago (in Astro 1.0). With our thriving developer community and so many new integrations being built all the time, it was finally time to fix this. We are pleased that this paves the way for even more integrations in the future!
```
OK, just a random attempt to put a bit of a story on this to add to the "finally fixed" narrative. We *could* position it with a nod to integrations, encourage people to build and keep building them etc. Just one idea of how that could look.

Reply:
It's a little bit long imo, but I'll play with it as a starting point